### PR TITLE
Allow team to add multiple admins

### PR DESCRIPTION
**Pull request by cp-mayank**
- Allow team to add multiple admins

[allow-multiple-admins.webm](https://github.com/user-attachments/assets/d2db5ce9-3bff-4208-a3cc-13a92fc6931b)

## Summary by CodeRabbit

- **New Features**
	- Added functionality for managing team administration, allowing users to promote team members to admin roles.
	- Introduced new localization strings for team admin features to enhance user experience.
	- Developed a new screen for making team admins with a reactive UI that integrates with state management.

- **Improvements**
	- Enhanced code readability and organization through streamlined variable declarations and formatting adjustments.
	- Improved error handling and state management within the team and player management processes.
	- Refined UI logic for displaying action buttons based on user roles and team membership.

- **Bug Fixes**
	- Resolved potential null reference issues in player data handling and improved conditional logic for user actions based on roles.

- **Documentation**
	- Updated localization files to reflect new strings related to team management functionalities.

### CODE REVIEW

File: khelo/lib/components/action_bottom_sheet.dart [hunk 1]
```diff
@@ -53,20 +53,22 @@ class BottomSheetAction extends StatelessWidget {
...
+    this.enabled = true,
     this.child,
     this.onTap,
   });
 
   @override
   Widget build(BuildContext context) {
     return OnTapScale(
-      enabled: child == null,
+      enabled: enabled,
```
Comment (cp-sidhdhi-p) [inline khelo/lib/components/action_bottom_sheet.dart:71]: have you managed other places where this shouldn't be enabled i.e more option in filter teams ??

File: data/lib/api/team/team_model.dart [hunk 2]
```diff
@@ -18,31 +20,38 @@ class TeamModel with _$TeamModel {
     String? profile_img_url,
     String? created_by,
     DateTime? created_at,
-    List<UserModel>? players,
+    @JsonKey(name: FireStoreConst.teamPlayers)
```
Comment (cp-sidhdhi-p) [inline data/lib/api/team/team_model.dart:23]: what does it do?

File: data/lib/service/team/team_service.dart [hunk 3]
```diff
@@ -27,99 +27,81 @@ class TeamService {
...
+              return player.copyWith(user: user);
+            }).toList();
+
+            return team.copyWith(players: players);
+          } else {
+            return null;
+          }
+        }),
+      );
+      return teams.whereType<TeamModel>().toList();
```
Comment (cp-sidhdhi-p) [inline data/lib/service/team/team_service.dart:104]: why do we need whereType?

File: khelo/lib/ui/flow/matches/match_detail/components/match_detail_squad_view.dart [hunk 4]
```diff
@@ -53,16 +53,18 @@ class MatchDetailSquadView extends ConsumerWidget {
...
+            .map((e) => e.user!)
             .toList() ??
         [];
 
     final secondTeamBench = secondTeam?.team.players
-            ?.where((element) =>
+            .where((element) =>
                 !secondTeamSquad.map((e) => e.id).contains(element.id) &&
-                element.isActive)
+                element.user!.isActive)
```
Comment (cp-sidhdhi-p) [inline khelo/lib/ui/flow/matches/match_detail/components/match_detail_squad_view.dart:66]: on what basis we're force unwrapping user? if we would manage user this way, we have to force unwrap or check for null  at every places which is not right thing to do as per my opinion

File: khelo/lib/ui/flow/team/detail/make_admin/make_team_admin_view_model.dart [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    try {
+      final players = _players.map((player) {
+        final role = state.selectedPlayers.contains(player)
+            ? TeamPlayerRole.admin
+            : TeamPlayerRole.player;
+        return player.copyWith(role: role);
+      }).toList();
+
+      await _teamService.editPlayersToTeam(_teamId, players);
+      state = state.copyWith(pop: true, actionError: null);
```
Comment (cp-sidhdhi-p) [inline khelo/lib/ui/flow/team/detail/make_admin/make_team_admin_view_model.dart:50]: change action error state to null at the start of function as it won't show error for back to back multiple fails.

**Comment by cp-sidhdhi-p** [comment]
@cp-mayank it allows to deselect all admin and save it, it shouldn't be like that, there must be single admin at a time, also always show save action button in makeAdminScreen, if it's disabled then show it with disabled color as we follow the same at every place.

adminRemoved

**Comment by cp-sidhdhi-p** [comment]
@cp-mayank previously we were checking for **team_created_by** to start match with selected team as we didn't had team admin, but now we need to check for admin, when user starts any match, he should be admin of at least one team also same applies when we navigate to add match from team detail screen with default selected screen. also filter for team should be changed as we need to show team whose admin is current user

AdminCanAddMatchOnly

File: data/lib/api/team/team_model.dart (hunk 2 above)
Comment (cp-mayank) [reply data/lib/api/team/team_model.dart:23]: It retrieves the value based on the Firebase key name, so we don't have to update the players field name in all places.

File: data/lib/service/team/team_service.dart (hunk 3 above)
Comment (cp-mayank) [reply data/lib/service/team/team_service.dart:104]: we are returning a null value in team list, we need to remove the null values, which is why we have used whereType.

File: data/lib/service/team/team_service.dart [hunk 6]
```diff
@@ -27,99 +27,81 @@ class TeamService {
...
-          id: teamRequestModel.id,
-          city: teamRequestModel.city,
-          created_at: teamRequestModel.created_at,
-          created_by: teamRequestModel.created_by,
-          profile_img_url: teamRequestModel.profile_img_url,
-          players: member,
-        );
-        return team;
-      }));
+    return _teamsCollection.snapshots().asyncMap((snapshot) async {
```
Comment (cp-sidhdhi-p) [inline data/lib/service/team/team_service.dart:81]: we're fetching all data and then sorting from it, it's not right way to do so, we can use data duplication for this, how about keeping players field as it is and adding another field called admins and it includes id of admins?

File: khelo/lib/ui/flow/team/detail/make_admin/make_team_admin_screen.dart [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  @override
+  Widget build(BuildContext context) {
+    _observePop();
+    _observeActionError();
+    final state = ref.watch(makeTeamAdminStateProvider);
+    return AppPage(
+      title: context.l10n.team_detail_make_admin,
+      actions: [
+        actionButton(context,
+            onPressed: notifier.onSave,
```
Comment (cp-sidhdhi-p) [inline khelo/lib/ui/flow/team/detail/make_admin/make_team_admin_screen.dart:51]: let's pass null id button is disabled

**Comment by cp-sidhdhi-p** [comment]
owner should be on top, other must be after that, also this spacing is not looking like figma

Simulator Screenshot - iPhone 15 - 2024-08-22 at 15 03 15

File: khelo/lib/ui/flow/team/detail/team_detail_screen.dart [hunk 8]
```diff
@@ -70,7 +71,11 @@ class _TeamDetailScreenState extends ConsumerState<TeamDetailScreen> {
             size: 24,
             color: context.colorScheme.textPrimary,
           )),
-      actions: (state.currentUserId == state.team?.created_by)
+      actions: (state.team?.created_by == state.currentUserId ||
+              state.team?.players.any((element) =>
+                      element.id == state.currentUserId &&
+                      element.role == TeamPlayerRole.admin) ==
+                  true)
```
Comment (cp-sidhdhi-p) [inline khelo/lib/ui/flow/team/detail/team_detail_screen.dart:78]: let's add it as property in team model so we can user it at multiple places

File: khelo/lib/ui/flow/team/add_team_member/add_team_member_view_model.dart [hunk 9]
```diff
@@ -50,7 +63,13 @@ class AddTeamMemberViewNotifier extends StateNotifier<AddTeamMemberState> {
   }
 
   void selectUser(UserModel user) {
-    state = state.copyWith(selectedUsers: [...state.selectedUsers, user]);
+    final player = TeamPlayer(
+        id: user.id,
+        role: (_team.created_by == currantUserId)
```
Comment (cp-sidhdhi-p) [inline khelo/lib/ui/flow/team/add_team_member/add_team_member_view_model.dart:68]: typo: currantUserId -> curr**e**ntUserId

**Review by cp-sidhdhi-p — APPROVED** [review]

**Pull request merged**